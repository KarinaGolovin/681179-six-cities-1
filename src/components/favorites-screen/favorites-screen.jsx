import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {toggleFavorite} from '../../store/actions';
import Favorites from '../favorites/favorites.jsx';
import {getFavoriteOffersByCities} from '../../store/reducers';
import PageTemplate from '../page-template/page-template.jsx';
import withAuthRedirect from '../../hocs/with-auth-redirect/with-auth-redirect';

const FavoritesWithRedirect = withAuthRedirect({
  notAuthorised: `/login`
})(Favorites);

export function FavoritesScreen(props) {
  const {favoriteList, updateBookmark} = props;
  return (
    <PageTemplate>
      <FavoritesWithRedirect favoriteList={favoriteList} onBookmarkClick={updateBookmark} />
    </PageTemplate>
  );
}

const mapStateToProps = (state) => {
  return {
    favoriteList: getFavoriteOffersByCities(state),
  };
};

const mapDispatchToProps = {
  updateBookmark: toggleFavorite
};

FavoritesScreen.propTypes = {
  favoriteList: PropTypes.array,
  updateBookmark: PropTypes.func,
};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(FavoritesScreen);

