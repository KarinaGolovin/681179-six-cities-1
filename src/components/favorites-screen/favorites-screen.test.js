import React from 'react';
import {FavoritesScreen} from './favorites-screen.jsx';
import ShallowRenderer from 'react-test-renderer/shallow';

const renderer = new ShallowRenderer();

const mock = [
  {
    title: `Test`,
    type: `Test`,
    price: 0,
    previewImage: ``,
    link: `#`,
    rating: 0,
    isPremium: true,
    id: 1,
  },
  {
    title: `Test`,
    type: `Test`,
    price: 0,
    previewImage: ``,
    link: `#`,
    rating: 0,
    isPremium: true,
    id: 2,
  },
];

it(`FavoritesScreen correctly renders`, () => {
  const component = renderer.render(
      <FavoritesScreen
        favoriteList={mock}
        updateBookmark={() => {}}
      />
  );

  expect(component).toMatchSnapshot();
});
