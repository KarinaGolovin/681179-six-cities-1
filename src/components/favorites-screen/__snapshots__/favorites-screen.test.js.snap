// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesScreen correctly renders 1`] = `
<Memo(ConnectFunction)>
  <Memo(ConnectFunction)
    favoriteList={
      Array [
        Object {
          "id": 1,
          "isPremium": true,
          "link": "#",
          "previewImage": "",
          "price": 0,
          "rating": 0,
          "title": "Test",
          "type": "Test",
        },
        Object {
          "id": 2,
          "isPremium": true,
          "link": "#",
          "previewImage": "",
          "price": 0,
          "rating": 0,
          "title": "Test",
          "type": "Test",
        },
      ]
    }
    onBookmarkClick={[Function]}
  />
</Memo(ConnectFunction)>
`;
